File: src/resources/nl-query.ts
import type { HttpClient } from '../http.js';
import type { NLQueryResult } from '../types/query.js';

export class NLQueryResource {
  constructor(private http: HttpClient) {}

  async ask(question: string): Promise<NLQueryResult> {
    return this.http.post<NLQueryResult>('/v1/nl-query', { question });
  }
}

File: python/invariance/resources/nl_query.py
from __future__ import annotations

from ..http_client import HttpClient
from ..types import NLQueryResult


class NLQueryResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def ask(self, question: str) -> NLQueryResult:
        return await self._http.post("/v1/nl-query", {"question": question})
